boa/src/builtins/number/mod.rs

```diff
@@ -405,6 +408,99 @@ impl Number {
         Ok(Self::to_number(this))
     }
 
+    pub(crate) fn parse_int(
+        _this: &mut Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        if args.len() > PARSE_INT_MAX_ARG_COUNT {
+            // Too many arguments.
+            return Err(Value::undefined());
+        }
+
+        if let (Some(val), r) = (args.get(0), args.get(1)) {
+            let mut radix = if let Some(rx) = r {
+                if let ValueData::Integer(i) = rx.data() {
+                    *i as u32
+                } else {
+                    // Handling a second argument that isn't an integer but was provided so cannot be defaulted.
+                    return Ok(Value::from(f64::NAN));
+                }
+            } else {
+                // No second argument provided therefore radix is unknown
+                0
+            };
+
+            match val.data() {
+                ValueData::String(s) => {
+                    // Attempt to infer radix from given string.
+
+                    if radix == 0 {
+                        if s.starts_with("0x") || s.starts_with("0X") {
+                            if let Ok(i) = i32::from_str_radix(&s[2..], 16) {
+                                return Ok(Value::integer(i));
+                            } else {
+                                // String can't be parsed.
+                                return Ok(Value::from(f64::NAN));
+                            }
+                        } else {
+                            radix = 10
+                        };
+                    }
+
+                    if let Ok(i) = i32::from_str_radix(s, radix) {
+                        Ok(Value::integer(i))
+                    } else {
+                        // String can't be parsed.
+                        Ok(Value::from(f64::NAN))
+                    }
+                }
+                ValueData::Integer(i) => Ok(Value::integer(*i)),
+                ValueData::Rational(f) => Ok(Value::integer(*f as i32)),
+                _ => {
+                    // Wrong argument type to parseInt.
+                    Ok(Value::from(f64::NAN))
+                }
+            }
+        } else {
+            // Not enough arguments to parseInt.
+            Err(Value::undefined())
+        }
+    }
+
+    pub(crate) fn parse_float(
+        _this: &mut Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        if args.len() > PARSE_FLOAT_MAX_ARG_COUNT {
+            // Too many arguments.
+            return Err(Value::undefined());
+        }
+
+        if let Some(val) = args.get(0) {
+            match val.data() {
+                ValueData::String(s) => {
+                    if let Ok(f) = f64::from_str(s) {
```

Comment:
I don't think the ::<64> can be avoided as otherwise, rust doesn't seem to be able to infer the type however I've removed FromStr + associated import.

The f64 import is kept to allow using f64::NAN.

Reply:
> The f64 import is kept to allow using f64::NAN

In theory, this is no longer needed since Rust 1.43:

https://doc.rust-lang.org/stable/std/primitive.f64.html#associatedconstant.NAN

Reply:
I updated my rust version and yeah it seems you are correct, fixed in latest commit :)